snuba_sdk/mql/mql.py

```diff
@@ -276,20 +284,32 @@ def visit_filter_factor(
         self, node: Node, children: Sequence[Any]
     ) -> Union[Condition, BooleanCondition]:
         factor, *_ = children
+
         # A nested filter can be both a boolean condition but also a single condition, since we allow `(condition)`.
         if isinstance(factor, BooleanCondition) or isinstance(factor, Condition):
             # If we have a parenthesized expression, we just return it.
             return factor
         else:
-            condition_op, lhs, _, _, _, rhs = factor
+            # condition_op, (variable or tag_key), space, colon, space, tag_value
+            condition_op, lhs, _, _, _, filter_factor_value = factor
+            contains_wildcard = filter_factor_value.contains_wildcard
+            rhs = filter_factor_value.value
             op = Op.EQ
-            if not condition_op and isinstance(rhs, list):
+            if not condition_op and contains_wildcard and isinstance(rhs, str):
+                if contains_wildcard:
+                    op = Op.LIKE
+
```

Comment:
nit: you could simplify it.
```suggestion
            if not condition_op and contains_wildcard and isinstance(rhs, str):
            	op = Op.LIKE
```

Reply:
Ah yeah that is in there twice, thank you for catching this!